File: Application/lib/models/organization.dart
class Organization {
  final String orgId;
  final String name;
  final String orgType;
  final String creditCode;
  final String description;
  final String? parentOrgId;
  final String creatorUserId;
  final DateTime createdAt;
  final String status;

  const Organization({
    required this.orgId,
    required this.name,
    required this.orgType,
    this.creditCode = '',
    this.description = '',
    this.parentOrgId,
    required this.creatorUserId,
    required this.createdAt,
    this.status = 'active',
  });

  bool get isSchoolClub => orgType == 'schoolClub';
  bool get isVolunteerTeam => orgType == 'volunteerTeam';
  bool get isSocialOrg => orgType == 'socialOrg';

  Map<String, dynamic> toJson() => {
        'orgId': orgId,
        'name': name,
        'orgType': orgType,
        'creditCode': creditCode,
        'description': description,
        if (parentOrgId != null) 'parentOrgId': parentOrgId,
        'creatorUserId': creatorUserId,
        'createdAt': createdAt.toIso8601String(),
        'status': status,
      };

  factory Organization.fromJson(Map<String, dynamic> json) => Organization(
        orgId: json['orgId'] as String,
        name: json['name'] as String,
        orgType: (json['orgType'] as String?) ?? 'schoolClub',
        creditCode: (json['creditCode'] as String?) ?? '',
        description: (json['description'] as String?) ?? '',
        parentOrgId: json['parentOrgId'] as String?,
        creatorUserId: (json['creatorUserId'] as String?) ?? '',
        createdAt: json['createdAt'] != null
            ? DateTime.tryParse(json['createdAt'] as String) ?? DateTime.now()
            : DateTime.now(),
        status: (json['status'] as String?) ?? 'active',
      );
}

File: CloudProgram/cloudfunctions/get-my-orgs/Organization.ts
export class Organization {
  orgId: string = '';
  name: string = '';
  orgType: string = 'schoolClub';
  creditCode: string = '';
  description: string = '';
  parentOrgId: string = '';
  creatorUserId: string = '';
  createdAt: Date = new Date();
  status: string = 'active';

  getClassName(): string {
    return 'Organization';
  }

  getFieldTypeMap(): Map<string, string> {
    const map = new Map<string, string>();
    map.set('orgId', 'String');
    map.set('name', 'String');
    map.set('orgType', 'String');
    map.set('creditCode', 'String');
    map.set('description', 'String');
    map.set('parentOrgId', 'String');
    map.set('creatorUserId', 'String');
    map.set('createdAt', 'Date');
    map.set('status', 'String');
    return map;
  }

  getPrimaryKeyList(): string[] {
    return ['orgId'];
  }

  getIndexList(): string[] {
    return ['orgId', 'name'];
  }

  getEncryptedFieldList(): string[] {
    return [];
  }

  static parseFrom(data: any): Organization {
    const obj = new Organization();
    if (data) {
      obj.orgId = data.orgId ?? '';
      obj.name = data.name ?? '';
      obj.orgType = data.orgType ?? 'schoolClub';
      obj.creditCode = data.creditCode ?? '';
      obj.description = data.description ?? '';
      obj.parentOrgId = data.parentOrgId ?? '';
      obj.creatorUserId = data.creatorUserId ?? '';
      obj.createdAt = data.createdAt ? new Date(data.createdAt) : new Date();
      obj.status = data.status ?? 'active';
    }
    return obj;
  }
}
